Plugins/Execution/Collect-Handles.ps1

```diff
@@ -0,0 +1,106 @@
+<#
+
+.SYNOPSIS
+    Plugin-Name: Collect-Handles.ps1
+    
+.Description
+
+	Collects handles information from remote systems for further analysis
+
+.EXAMPLE
+
+	StandAlone Execution
+
+	.\Collect-Handles.ps1 -ComputerName Test-PC
+
+	Power-Response Execution
+
+	set ComputerName Test-PC
+	run
+
+.NOTES
+    Author: Drew Schmitt
+    Date Created: 3/16/2019
+    Twitter: @5ynax
+    
+    Last Modified By:
+    Last Modified Date:
+    Twitter:
+  
+#>
+
+param (
+
+    [Parameter(Mandatory=$true,Position=0)]
+    [string[]]$ComputerName
+
+    )
+
+process {
+
+	#Verify that bin dependencies are met
+    $Bin_32 = ("{0}\handle.exe" -f $global:PowerResponse.Config.Path.Bin)
+    $Bin_64 = ("{0}\handle64.exe" -f $global:PowerResponse.Config.Path.Bin)
+
+    if (!(Test-Path $Bin_32)){
+
+        Throw "handle.exe not found in {0}. Place executable in binary directory and try again." -f $global:PowerResponse.Config.Path.Bin
+    }
+
+    if (!(Test-Path $Bin_64)){
+
+        Throw "handle64.exe not found in {0}. Place executable in binary directory and try again." -f $global:PowerResponse.Config.Path.Bin
+    }
+
+    foreach ($Computer in $ComputerName){
+
+        #Establish PowerShell Session
+        $Session = New-PSSession -ComputerName "$Computer" -SessionOption (New-PSSessionOption -NoMachineProfile)
+
+        #Determine remote system architecture
+        $OSArchitecture = Invoke-Command -Session $Session -ScriptBlock {(Get-WmiObject Win32_OperatingSystem).OSArchitecture}
+
+        #Select the proper EXE for OS Architecture
+        if ($OSArchitecture -eq "64-bit") {
+
+            $LocalExe = $Bin_64
+            $RemoteExe = "handle64.exe"
+        }
+
+        if ($OSArchitecture -eq "32-bit") {
+
+            $LocalExe = $Bin_32
+            $RemoteExe = "handle.exe"
+        }
+
+        #Copy binary to remote machine for execution and data collection
+        try {
+
+            Copy-Item -Path $LocalExe -Destination "C:\ProgramData\$RemoteExe" -ToSession $Session -Force -ErrorAction Stop
+
+        } catch {
+
+            throw "Error copying the binary to the remote maching. Quitting."
+        }
+
+        #Collect Data from the remote machine
+
+        $HandlesRaw = Invoke-Command -Session $Session -ScriptBlock {Invoke-Expression -Command "C:\ProgramData\$($args[0]) -accepteula | Select-Object -Skip 5"} -ArgumentList $RemoteExe
+
+        #Format data for ingestion into readable Format
+
+        $Handles = $HandlesRaw | Select-Object -Property @{ N = "Process"; E = { ($_ -split '(?<!:)\s+')[0] } }, @{ N = "ID"; E = { (($_ -split '(?<!:)\s+')[1] -replace 'pid:\s*') } }, @{ N = "Type"; E = { ($_ -split '(?<!:)\s+')[2] -replace 'type:\s*' } }, @{ N = "Path"; E = { ($_ -split '(?<!:)\s+')[3] -replace '^[^ ]*: ' } }
```

Comment:
This is an interesting way to create custom attributes! I'm not sure exactly how it goes or what form `$Handles` takes after the `Select-Object`, but the output is missing all of the `Process` column. In looking at the raw `handle.exe` output you might need to do some funky `-split '-{5,}` to split the string output into the different process chunks based on the `------------...` separators

Reply:
decided to keep as it.